engine-protocol/src/main/proto/engine_control.proto

```diff
@@ -34,15 +34,27 @@ message WaitForIdleStateResponse {}
 message GetRecordsRequest {}
 
 message RecordResponse {
+  // A JSON representation of a Record.
   string recordJson = 1;
 }
 
 service EngineControl {
 
+  /*
+    Starts the in-memory engine.
+   */
   rpc StartEngine (StartEngineRequest) returns (StartEngineResponse);
 
+  /*
+    Stops the in-memory engine.
+   */
   rpc StopEngine (StopEngineRequest) returns (StopEngineResponse);
 
+  /*
+    Resets the in-memory engine. This is done by recreating the engine entirely.
+    The engine should be reset before executing the next test. This prevents
+    the tests from influencing each other.
+   */
   rpc ResetEngine (ResetEngineRequest) returns (ResetEngineResponse);
```

Comment:
👍 I like it!

🔧 I seems to me that there is still a chance for tests to influence each other. If 2 tests run concurrently, and 1 resets the engine, it affects the other test. Right? If so, please make it clear from this doc that the user should still watch out for such situations.